Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -5,12 +5,47 @@ import Foundation
 
 /// A class to facilitate running code generation
 public class ApolloCodegen {
+  public enum Error: Swift.Error, LocalizedError {
+    case validationFailed(atLines: [String])
+
+    public var errorDescription: String? {
+      switch self {
+      case let .validationFailed(lines):
+        return "Schema and Operation validation failed! Check \(lines)"
+      }
+    }
+  }
   /// Executes the code generation engine with a specified configuration.
   ///
   /// - Parameters:
   ///   - configuration: The configuration to use to build the code generation.
   public static func build(with configuration: ApolloCodegenConfiguration) throws {
     try configuration.validate()
+
+    let compilationResult = try compileResults(using: configuration.input)
+    #warning("TODO - compilationResult will be passed into the next step")
+  }
+
+  static func compileResults(
+    using input: ApolloCodegenConfiguration.FileInput
+  ) throws -> CompilationResult {
+    let frontend = try GraphQLJSFrontend()
+
+    let schemaURL = URL(fileURLWithPath: input.schemaPath)
+    let graphqlSchema = try frontend.loadSchema(from: schemaURL)
+
+    let matches = try Glob(input.searchPaths).match()
+    let documents = try matches.map({ path in
+      return try frontend.parseDocument(from: URL(fileURLWithPath: path))
+    })
+    let mergedDocument = try frontend.mergeDocuments(documents)
+
+    let graphqlErrors = try frontend.validateDocument(schema: graphqlSchema, document: mergedDocument)
+    guard graphqlErrors.isEmpty else {
+      throw Error.validationFailed(atLines: graphqlErrors.flatMap({ $0.logLines }))
```

Comment:
This can be a project for another day but I'd like us to log the actual graphQL errors here instead of just "Failed, check lines 134 & 178". Can we create an issue for 1.0 to improve this please?

Reply:
Are you meaning log via `CodegenLogger`? That's easy enough to add into this PR.

Something up for debate is whether we should throw `GraphQLError`s as I've got here. graphql-js doesn't throw but IMO if there are validation errors _we_ should halt code generation until they're resolved. Happy to be convinced otherwise if there is a strong opposing argument.

Reply:
I've added logging of these errors in https://github.com/apollographql/apollo-ios/pull/2055/commits/8e624b6d5633bc3e78889a2418448014b5c81597.

There are two parts to your comment:
> log the actual graphQL errors here instead of just "Failed, check lines 134 & 178"

1. The `logLines` property on GraphQLError isn't simply a line number indicator. It is instead a combination of all relevant parts that constitute the error, both for validation and syntax errors - see [here](https://github.com/apollographql/apollo-ios/blob/release/1.0-alpha-incubating/Sources/ApolloCodegenLib/Frontend/GraphQLError.swift#L37).
2. Console logging for useful debugging output.

So, this change will result in console messages like below with the same being thrown (the relevant error message I think you're looking for is at the end):
```
[ERROR - ApolloCodegenLib:ApolloCodegen.swift:46] - ["/Users/calvincestari/Projects/apollo-ios/Tests/ApolloCodegenTests/Output/Codegen/6B8FF0BF-AB8A-4B15-B697-04CDBDA23B86.graphql:4:error:Cannot query field \"name\" on type \"Book\"."]
```